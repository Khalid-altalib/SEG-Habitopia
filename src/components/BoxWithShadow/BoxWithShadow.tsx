// React Native
import { View } from "react-native";

export type BoxWithShadowProps = {
  /** The components to show inside the box. */
  children: any;
};

/**
 * A box that has a shadow and can hold components inside it.
 *
 * @param props - The properties passed to the component.
 * @returns - The box.
 */
const BoxWithShadow = ({ children }: BoxWithShadowProps) => (
  <View
    style={[
      {
        shadowColor: "black",
        shadowOffset: { width: 5, height: 5 },
        shadowOpacity: 0.15,
        borderRadius: 10,
      },
    ]}
  >
    {children}
  </View>
);

export default BoxWithShadow;
