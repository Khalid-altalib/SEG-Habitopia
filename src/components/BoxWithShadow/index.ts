import BoxWithShadow from "./BoxWithShadow";
export default BoxWithShadow;
