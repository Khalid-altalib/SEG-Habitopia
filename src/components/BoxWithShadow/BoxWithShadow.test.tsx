// React
import React from "react";

// React Native Testing Library
import { render, screen } from "@testing-library/react-native";

// Habitopia
import BoxWithShadow from "@components/BoxWithShadow";

it("Renders correctly.", () => {
  render(<BoxWithShadow>This is some example content.</BoxWithShadow>);

  expect(screen.toJSON()).toMatchSnapshot();
});
