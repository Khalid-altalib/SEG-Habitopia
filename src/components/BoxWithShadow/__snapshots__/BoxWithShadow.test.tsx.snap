// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly. 1`] = `
<View
  style={
    [
      {
        "borderRadius": 10,
        "shadowColor": "black",
        "shadowOffset": {
          "height": 5,
          "width": 5,
        },
        "shadowOpacity": 0.15,
      },
    ]
  }
>
  This is some example content.
</View>
`;
